File: server/rest/pkg/data/user_repo.go
package data

import (
	"github.com/pocketbase/dbx"
	"github.com/runshop/server/rest/pkg/models"
)

type IUserRepo interface {
	IBaseRepo[models.User]
}

type UserRepo struct {
	BaseRepo[models.User]
}

func NewUserRepo(dbContext *dbx.DB) IUserRepo {
	return &UserRepo{
		BaseRepo[models.User]{
			db:        dbContext,
			tableName: "users",
		},
	}
}

File: server/rest/runshop.rest/runShop.data/Repos/user/UserRepo.cs
using runShop.data.Repos.baseRepo;
using runShop.Models.models;

namespace runShop.data.Repos.user
{
    public class UserRepo : BaseRepo<User>, IUserRepo
    {
        public UserRepo(AppDbContext appDbContext) : base(appDbContext)
        {
        }
    }
}

